test/e2e/linuxbridge_test.go

```diff
@@ -0,0 +1,106 @@
+package e2e
+
+import (
+	"context"
+	"io/ioutil"
+	"testing"
+	"time"
+
+	yaml "github.com/ghodss/yaml"
+
+	apis "github.com/nmstate/kubernetes-nmstate/pkg/apis"
+	nmstatev1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	createBridgeCr       = "deploy/crds/create-br1-linux-bridge.yaml"
+	deleteBridgeCr       = "deploy/crds/delete-br1-linux-bridge.yaml"
+	firstNodeName        = "node01" // TODO: Get it from cluster
+	bridgeName           = "br1"
+)
+
+func TestLinuxBridge(t *testing.T) {
+
+	nodeNetworkStateList := &nmstatev1.NodeNetworkStateList{}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, nodeNetworkStateList)
+	if err != nil {
+		t.Fatalf("failed to add custom resource scheme to framework: %v", err)
+	}
+
+	ctx := framework.NewTestCtx(t)
+	defer ctx.Cleanup()
+
+	err = ctx.InitializeClusterResources(&framework.CleanupOptions{TestContext: ctx, Timeout: cleanupTimeout, RetryInterval: cleanupRetryInterval})
+	if err != nil {
+		t.Fatalf("failed to initialize cluster resources: %v", err)
+	}
+
+	// get namespace
```

Comment:
this one is obvious, but could you comment on previous blocks? what's going on there. the same for the rest of the test. this whole function is really long, please comment each step of the test, what are you doing and why, what are your expectations. i find reading go code really hard, full of error checking and helper functions.

Reply:
I am going to conver this to ginkgo (now I am not so hater :-).